src/components/image/index.js

```diff
@@ -88,6 +96,25 @@ class Image extends PureBaseComponent {
       />
     );
   }
+
+  onLayout = (event) => {
+    const width = event.nativeEvent.layout.width;
+    this.setState({width});
+  }
+
+  render() {
+    const {style, overlayType} = this.getThemeProps();
+  
+    if (overlayType) {
+      return (
+        <View onLayout={this.onLayout}>
+          {this.renderImage()}
+          <Overlay style={[style, {width: this.state.width}]} type={overlayType}/>
```

Comment:
Passing the same style you pass to the image might be risky here. 
I wouldn't pass anything else to the overlay and keep is simple at this point.
but if we do want to support custom style for the overlay, we need a separate prop.

Reply:
I'm passing the style to position the overlay properly on the image. Not for actual styling purposes.